sinatra/sinatra_app.rb

```diff
@@ -67,6 +76,51 @@
   dataset[id: id].to_json
 end
 
+post '/api/templates/:id/delete' do
+  protected!
+
+  id = params[:id].to_i
+  dataset = database[:templates]
+  dataset.filter(id: id).delete
+end
+
+post '/api/templates' do
+  protected!
+
+  data = JSON.parse(request.body.read)
+
+  example_activities = "{}"
+  if data['example_activities'].any?
+    example_activities = Sequel.pg_array(data['example_activities'])
+  end
+
+  req_certifications = "{}"
+  if data['req_certifications'].any?
+    req_certifications = Sequel.pg_array(data['req_certifications'])
+  end
+
+  dataset = database[:templates]
+  dataset.insert(
+    job_title: data['job_title'],
+    company_description: data['company_description'],
+    job_description: data['job_description'],
+    req_occupational_skills: Sequel.pg_json(data['req_occupational_skills']),
+    req_foundational_skills: Sequel.pg_json(data['req_foundational_skills']),
+    pref_occupational_skills: Sequel.pg_json(data['pref_occupational_skills']),
+    pref_foundational_skills: Sequel.pg_json(data['pref_foundational_skills']),
+    example_activities: example_activities,
+    req_certifications: req_certifications
+  )
+end
+
+# serve secured section for 'manage' folder
+get '/manage' do
+  protected!
+  file_name = "_site#{request.path_info}/index.html".gsub(%r{\/+}, '/')
+  raise Sinatra::NotFound unless File.exist?(file_name)
+  File.read(file_name)
+end
```

Comment:
Accessing this page & logging in should be documented in the README. I had to come in here and look for the route to remember what url to go to.